### ISSUE TITLE
Please add support for CzechRad CZRA1 header code

### ISSUE THREAD

**Issue by juhele** [Safecast/safecastapi#1034]
https://github.com/Safecast/safecastapi/blob/6cf3612427b8d05c4855d491646f40da74a16987/app/models/bgeigie_import.rb#L179

First prototype units used CZRDD, current CzechRads use CZRA1, in future possibly others like CZRA2, CZRB1 - this is because in case of commercial production of CzechRads there might be device number assignment conflicts if we had only CZRDD code.

thanks

References:
https://github.com/juhele/CzechRad/tree/main/DATA

**Comment by robouden** [comment]
Jan,

To make the change in the code would be easy. But to get it deployed and in the app, we do not currently have anyone who is confident in doing that and who can allocate time for doing that. I would suggest making a change to use the old header till we can add your new header to the log file.

Regards
Rob

**Comment by juhele** [comment]
oops,
I think we will have to wait because we already have several hundreds of CzechRads in the field and there is no way to do such change remotely...

the only way would be to change it in every LOG file... (probably possible with some batch script) but reuploading those 700 LOGs looks like a nightmare...

**Comment by robouden** [comment]
Yes, we notice about 750 (according to Pieter). It would have been helpful if you had tested one upload only and see if that worked. BTW.. For the changing of the header in  the batch file, you have to recalculate the chksum too.

Regards
Rob

**Comment by juhele** [comment]
I had the same issue at the beginning with the prototype data with old header so I thought you did not yet implemented it. Later, during uploading I noticed that old data is imported correctly but new not...

The checksum is not problem if I load and save the LOG from QGIS. However, both data types are displayed correctly in the API. 
In addition the Tilemap is still not being updated yet so we have to wait anyway...

**Comment by robouden** [comment]
Jan,

For getting drives approved, you need a correct checksum.
I tried with a file you had uploaded and only changed the header string. It
did upload but could not be processed/approved due to checksum errors.

Regards
Rob

**Comment by juhele** [comment]
RadiationToolbox handles the checksum when saving the LOG file. So you recommend to manually change the newly uploaded CzechRad LOGs to "legacy" header CZRDD to temporarily solve this issue? (until properly implemented) PS: I don't think it's within my time to redo the previously uploaded files as new measurements keep pouring in... And I think it it quite important to manage the quality control.

**Comment by robouden** [comment]
Jan,

I will try one file and report back to you if that setup works. Please wait.

Regards
Rob

**Comment by robouden** [comment]
Jan,

Just tested it and your idea works, see: [https://api.safecast.org/en-US/bgeigie_imports/64632](https://api.safecast.org/en-US/bgeigie_imports/64632
)
Hugs and peace,
Rob

**Comment by robouden** [comment]
Jan,

I think you can also script a batch that can upload directly (from https://api.safecast.org):

![image](https://github.com/user-attachments/assets/e97de8ef-64f3-471c-8f06-8948f6f8acf0)

**Comment by juhele** [comment]
OK,
a temporary solution consisting of manual conversion to legacy CzechRad code seems to work:
https://api.safecast.org/en-US/bgeigie_imports?by_user_id=6776&order=created_at+desc
I will use it for new data files, but I hope that after implementation into the API it will be possible to load (previously uploaded) files with the new CzechRad code in the future...

**Comment by robouden** [comment]
Jan,

Thanks for using the temporal solution.
I also hope we can fix the issue soon.

Regards
Rob

**Comment by robouden** [comment]
Jan,

Sorry to bother you. I have been trying to upload without the GUI log
files, but I have not managed it yet.
We were wondering if your team upload the log files through a POST or
manually with login user/password?


On the api.safecast.org is a mention how to do a POST to the API for
import. See below screenshot.

[image: image.png]

Hugs and peace,
Rob
